ci/prepare_environment.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+if git whatchanged --name-only --pretty="" master...HEAD | grep "Dockerfile\|.dockerignore"; then
+  echo "Dockerfile changes detected..."
+  docker build --rm -t "${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHA}" .
+  docker run --rm "$CI_COMMIT_SHA:$CI_PIPELINE_ID" "/test/buildenv_check.sh"
+else
```

Comment:
1. No need for an 'else' here just 'exit 0' in the if statement. 
2. And always end your script with and exit statement, very important, otherwise you can just do: 
`echo "malicious script code.. >> yourscript.sh   `
3. Always an empty line at the end of the scripts. 

if ....; then
...
exit 0
fi
....
....
exit 0